staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/strategy.go

```diff
@@ -143,8 +210,19 @@ func (a customResourceValidator) Validate(ctx genericapirequest.Context, obj run
 	if !ok {
 		return field.ErrorList{field.Invalid(field.NewPath(""), customResourceObject, fmt.Sprintf("has type %T. Must be a pointer to an Unstructured type", customResourceObject))}
 	}
-
 	customResource := customResourceObject.UnstructuredContent()
+
+	if utilfeature.DefaultFeatureGate.Enabled(apiextensionsfeatures.CustomResourceSubResources) {
```

Comment:
These tests should go into the CRD validation.

Reply:
also below

Reply:
This validates the "custom resource". So I think it should not go in the CRD strategy and should instead remain the in the CR strategy.

@sttts WDYT?

Reply:
hmm, I think `customResource["oneOf"]` does not make sense here. We have to check that OneOf is not used in the root of the JSON schema in the CRD.